src/core/dash.h

```diff
@@ -308,8 +312,8 @@ class DashTable : public detail::DashTableBase {
   // the same object. IterateDistinct goes over all distinct segments in the table.
   template <typename Cb> void IterateDistinct(Cb&& cb);
 
-  auto EqPred() const {
-    return [p = &policy_](const auto& a, const auto& b) -> bool { return p->Equal(a, b); };
+  template <typename K> auto EqPred(const K& key) const {
+    return [p = &policy_, &key](const auto& probe) -> bool { return p->Equal(probe, key); };
```

Comment:
and p->Equal is just ==, ironically we never use something else with our multiple layers of indirection... I think it can be simplified, but that's out of scope for this PR as it would erase large parts of code